lampod/src/lib.rs

```diff
@@ -167,16 +181,88 @@ impl LampoDaemon {
 
     pub fn init_peer_manager(&mut self) -> error::Result<()> {
         log::debug!(target: "lampo", "init peer manager ...");
-        let mut peer_manager = LampoPeerManager::new(&self.conf, self.logger.clone());
+        let mut peer_manager;
+        let peer_liquidity;
+
+        if let Some(liq) = &self.liquidity {
+            peer_liquidity = Some(Arc::clone(&self.liquidity.clone().unwrap()));
+        } else {
+            peer_liquidity = None;
+        }
```

Comment:
```suggestion
        let peer_manager = if let Some(liq) = &self.liquidity {
            peer_liquidity = Some(Arc::clone(&self.liquidity.clone().unwrap()));
        } else {
            peer_liquidity = None;
        }
```

use this syntax, the one that you are using is really unusual to rust